### PR TITLE
[WIP] HLA-1096: Added Reproject function

### PR DESCRIPTION
**Pull request by s-goldman**
Resolves [HLA-1096](https://jira.stsci.edu/browse/HLA-1096)

This PR adds the reproject function to the codebase. This code was based on a similar function for JWST. A main difference between the two is that this code assumes that we are using the high-level [WCS API](https://docs.astropy.org/en/latest/wcs/index.html#loading-wcs-information-from-a-fits-file). Several unit tests are also added for the case of inputted strings, array, as well as an expected failure. 

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [x] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [ ] added relevant label(s)

### CODE REVIEW

**Comment by s-goldman** [comment]
I still need to add the relevant documentation.

**Comment by nden** [comment]
@s-goldman Please make the PR against the `stcal-alignment` branch, not `main`. We ahven't merged that into main yet.

**Comment by s-goldman** [comment]
> @s-goldman Please make the PR against the `stcal-alignment` branch, not `main`. We ahven't merged that into main yet.

Thanks. Switch made. I still need to add documentation and possibly a regression test.

File: tests/current_gdqfits
Comment (kmacdonald-stsci) [inline tests/current_gdqfits:1]: What is this file?

File: src/stcal/alignment/reproject.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        --------
+        tuple
+            Tuple of reprojected x and y coordinates.
+        """
+        # example inputs to resulting function (12, 13, 0) # third number is origin
+        if not isinstance(x, list):
+            x = [x]
+        if not isinstance(y, list):
+            y = [y]
+        print(x)
```
Comment (kmacdonald-stsci) [inline src/stcal/alignment/reproject.py:82]: Remove this print or use logger.

**Pull request closed without merging**